src/main/resources/assets/js/app/view/context/WidgetsSelectionRow.ts

```diff
@@ -118,3 +124,50 @@ export class WidgetViewOption {
     }
 
 }
+
+export class WidgetViewer extends NamesAndIconViewer<WidgetViewOption> {
+
+    constructor() {
+        super('widget-viewer');
+    }
+
+    doLayout(object: WidgetViewOption) {
+        super.doLayout(object);
+
+        const view = this.getNamesAndIconView();
+        if (object && object.getWidgetView() && view) {
+            const widgetClass = object.getWidgetView().getWidgetKey() != null ? 'external-widget' : 'internal-widget';
+            view.removeClass('external-widget internal-widget');
+            view.addClass(widgetClass);
+        }
+    }
+
+    resolveDisplayName(object: WidgetViewOption): string {
+        return object.getWidgetView().getWidgetName();
+    }
+
+    resolveSubName(object: WidgetViewOption): string {
+        const name = object.getWidgetView().getWidgetName();
+        const description = object.getWidgetView().getWidgetDescription() || 'No description';
+        switch (name) {
+        case 'Properties': return 'Properties of the content';
+        case 'Version history': return 'History of the content\'s changes';
+        case 'Dependencies': return 'Dependencies tree of the content';
+        default:
```

Comment:
You cannot rely on hardcoded widget names to resolve description or icon because they are localised. Try switching your browser from English to any other language and it will stop working. We need something like widgetId for internal widgets that won't depend on browser language.